docs/prompt-api.md

```diff
@@ -84,6 +84,131 @@ fun main() {
 }
 ```
 
+## Multimodal inputs
+
+In addition to providing text messages within prompts, Koog also lets you send images, audio, video, and files to LLMs along with `user` messages. As with standard text-only prompts, you also add media to the prompt using the DSL structure for prompt construction.
+
+```kotlin
+val prompt = prompt("multimodal_input") {
+    system("You are a helpful assistant.")
+
+    user {
+        +"Describe these images"
+
+        attachments {
+            image("https://example.com/test.png")
+            image(Path("/User/koog/image.png"))
+        }
+    }
+}
+```
+
+### Textual prompt content
+
+To accommodate for the support for various attachment types and create a clear distinction between text and file inputs in a prompt, you put text messages in a dedicated `content` parameter within a user prompt. 
+To add file inputs, provide them as a list within the `attachments` parameter. 
+
+The general format of a user message that includes a text message and a list of attachments is as follows:
+
+```kotlin
+user(
+    content = "This is the user message",
+    attachments = listOf(
+        // Add attachments
+    )
+)
+```
+
+### File attachments
+
+To include an attachment, provide the file in the `attachments` parameter, following the format below:
+
+```kotlin
+user(
+    content = "Describe this image",
+    attachments = listOf(
+        Attachment.Image(
+            content = AttachmentContent.URL("https://example.com/capture.png"),
+            format = "png",
+            mimeType = "image/png",
+            fileName = "capture.png"
+        )
+    )
+)
+```
+
+The `attachments` parameter takes a list of file inputs, where each item is an instance of one of the following classes:
+
+- `Attachment.Image`: local image files or images from a URL in various image file formats.
```

Comment:
A small suggestion - let's remove content type mentions, these are explained in more details separately. Let's just put something like 
> `Attachment.Image`: image attachments, e.g. `jpg`, `png` 
> `Attachment.Audio`: audio attachments, e.g. `mp3`, `wav`
> etc.